### ISSUE TITLE
Did you use Drop Out?

### ISSUE THREAD

**Issue by rlaengud123** [doronharitan/human_activity_recognition_LRCN#2]
In the paper(https://arxiv.org/pdf/1411.4389.pdf, 5page)

They said "Additionally,
when training the LRCN network end-to-end, we found that
aggressive dropout (0.9) was needed to avoid overfitting"

But, I cant find Drop Out.

Did you use Drup Out In your code?

Thx

**Comment by doronharitan** [comment]
Hi, 
I didn't use dropout. I stop the training of the model before the overfitting (used the train vs validation learning curve do decide when to do so)
Also, as you can see I tested the model on new data (not even the test data) and its performance was over 91% so I figure that it doesn't overfit.

**Comment by rlaengud123** [comment]
Thank to your answer.
I have some questions..
 
1.
I used your code well (thank you).
but, end the test, i got acc is 55%.
I think my parameter was wrong.
can you share the parameter?
 
2.
I want to use Multi-GPU.
I have 8 GPUs, but when i use train.py, workstation used only device:0 2GB(i think in your code "cuda:device:0")
In torch, offer DataParallel, Can I use it? or in your code offer Multi-GPU?
How can i use Mulit-GPU? Thx
-----Original Message-----

**Comment by doronharitan** [comment]
Hi,
1. as I mentioned in the readme file I trained and tested my model on 55 categories from the UCF101 possible categories. 
My params were:
Preprocessing step:
--sampling_rate             10          
--num_frames_to_extract     5
--ucf101_fps                25

train.py params
--epochs                    40
--batch-size                256
--lr                              5e-4
--split_size                0.2
--seed                      42
--latent_dim                512
--hidden_size               256 
--lstm_layers               2
--bidirectional             True
--number_of_classes         55

test mode params:
--num_frames_video          5
--seed                      42
--latent_dim                512
--hidden_size               256
--lstm_layers               2
--bidirectional             True
--number_of_classes         55
--model_dir                 model_dir   
--model_name                model_name  

2. I don't have experience with multi-GPU training (except GCP).
I trained my model on Google Cloud Platform using K80 GPU, took ~ 25 hours to reach ~93% accuracy on validation set

Hope this helps. 
Good luck
Doron

**Comment by rlaengud123** [comment]
Sorry, I have another question.

Are there any rules for trainlist (1,2,3) and testlist (1,2,3)?

Is it simply random?

However, I see some repeating sections. And also regularity in the file.

When you run the code
Are all three sets used?

How should I organize my files?

Thank you, sir.
ᐧ

2020년 2월 6일 (목) 오전 1:16, doronharitan <notifications@github.com>님이 작성:

**Comment by yangyuan6956873** [comment]
hi,I have the same question.
If I want to train two categories from the UCF101 or other datasets.
How can i change the trainlist and the testlist.
Thank you very much.
